### PR TITLE
Add basic filtering and tracking setup

### PR DESCRIPTION
**Pull request by mschauer**

### CODE REVIEW

File: src/filter.jl [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+function trackandfilter(M, imgs, X0, P0, params...)
+    nframe = size(imgs, 3)
+    pp = fill(zero(M.x0), nframe, length(X0))
+
+    for k in 1:length(X0)
+        x = X0[k]
+        P = P0[k]
+
+        for t in 1:nframe
+            x, Ppred, A = Kalman.predict!(t, x, P, t + 1, M)
```
Comment (yakir12) [inline src/filter.jl:19]: Here you are `predict!`ing on frame # 1, how?

Comment (yakir12) [reply src/filter.jl:19]: but it seems like the initial guesses are not going to function so well as a "previous" estimate to where the root was one frame ago, wouldn't it be better to start predicting on frame #2?

Comment (mschauer) [reply src/filter.jl:19]: This follows their trajectories assuming the current position and speed.

File: test/testfilter.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        ((574.405, 539.448), 1.83611),
+        ((693.938, 521.589), 1.82364),
+        ((694.327, 356.419), 5.10786),
+        ((798.665, 635.654), 2.35362),
+        ((884.091, 460.284), 2.18453),
+        ((948.329, 593.023), 2.55287)
+]
+M = Model1
+ntips = length(tips)
+X0 = [SVector(tips[i][1]..., M.x0[3:4]...) for i in 1:ntips]
```
Comment (mschauer) [inline test/testfilter.jl:31]: The model is initalized with starting positions on the tips and their starting speeds here. Starting speed is taken from `M.x0` set in `models.jl`

File: src/miscsetup.jl [hunk 3]
```diff
@@ -0,0 +1,8 @@
+using Images, Kalman, GaussianDistributions
+
+import Base.LinAlg.normalize!
+function normalize!(x::AbstractArray)
+    l, u = extrema(x)
+    (x - l)/(u - l)
+end
+greyscale(A) = map(x->RGB(x,x,x), normalize!(copy(A)));
```
Comment (yakir12) [inline src/miscsetup.jl:8]: Is the purpose of these function just for the presentation or is it *actually* important for the Kalman filter to get values that are between 0 and 1?

File: src/filter.jl (hunk 1 above)
Comment (yakir12) [reply src/filter.jl:19]: but how can we get an estimate for their speed from frame 1? Oh wait, we need to manually give the estimate, right?

File: src/miscsetup.jl (hunk 3 above)
Comment (mschauer) [reply src/miscsetup.jl:8]: No, the filter does not see the image, it *only* sees it through the calls of `track`.

Comment (yakir12) [reply src/miscsetup.jl:8]: OK, so there's no real need for them in the sense of the filter.

File: test/testfilter.jl [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+using Base.Test
+
+sz = (1024, 1024)
+nframe = 25
+@time imgs0 = zeros(sz...,  nframe)
+@time for k in 1:nframe
+    imgs0[:,:,k] = convert(Matrix{Float64}, Images.load(joinpath(Pkg.dir("TrackRoots"), "test", "testimage", "$k.TIF")))
+end
+imgs = normalize!(imgs0[1:768, :, :])
```
Comment (mschauer) [inline test/testfilter.jl:14]: To normalise or not to normalise here only affects the error estimate in `track` which we do not use yet.

File: test/testfilter.jl [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        ((884.091, 460.284), 2.18453),
+        ((948.329, 593.023), 2.55287)
+]
+M = Model1
+ntips = length(tips)
+X0 = [SVector(tips[i][1]..., M.x0[3:4]...) for i in 1:ntips]
+P0 = [M.P0 for i in 1:ntips]
+roots = trackandfilter(tostatic(Model1), imgs, X0, P0, 10)
+
+@test norm(roots[end][1:2] - [930.207, 692.298]) < 0.1
```
Comment (mschauer) [inline test/testfilter.jl:35]: This just checks that the last tip ends at a certain location 930.207, 692.298 where I think it should be.

**Pull request merged**